### ISSUE TITLE
Can you please release the code even if it is in Theano?

### ISSUE THREAD

**Issue by sverma88** [A2Zadeh/MARN#4]

**Comment by A2Zadeh** [comment]
Hi @sverma88. Yes, I have been working on this for the past few days. Let me get back to you by the end of weekend. In the meantime, would the MARN fusion code in the SDK help you? Or would you rather just wait a few days to get the theano code.

**Comment by sverma88** [comment]
Hi Amir,

The code is SDK will be of huge help, but is it available?
I'm not able to find it, can you specify the path here?

Thanks

**Comment by A2Zadeh** [comment]
@sverma88 Of course. Here is the fusion layer: https://github.com/A2Zadeh/CMU-MultimodalSDK/blob/master/mmsdk/mmmodelsdk/fusion/multiple_attention/model.py

Implementing LSTHM should be also straightforward but here is the implementation we have: https://github.com/A2Zadeh/CMU-MultimodalSDK/tree/master/mmsdk/mmmodelsdk/modules/LSTHM

I am going to share the code for theano for MFN, MARN and Graph-MFN as soon as I can. I just need to zip the folders which can be a bit big.

**Comment by sverma88** [comment]
@A2Zadeh Thanks alot.

**Comment by A2Zadeh** [comment]
@sverma88 Of course. Happy to help. Just an update, it is taking a long time to compress. The code is around 400GB with all the data. I think it will take up another day. And then I will put it on my google drive, so I think the whole process realistically will take up the entire weekend.

**Comment by sverma88** [comment]
@A2Zadeh please let me know whenever the process it complete. 
At this moment, the code is a big bottleneck for me!

**Comment by A2Zadeh** [comment]
@sverma88 The zipping is just done. I had to kill the previous zipping and shrink the size and remove some stuff that was not needed. It is now being uploaded to my gdrive. I will share the link in a few hours.

**Comment by A2Zadeh** [comment]
@sverma88 Here is the download link:

https://drive.google.com/drive/folders/1NxyFuogyzNFoCH0Zi5aIXUGYKGuSY9TY?usp=sharing

**Comment by sverma88** [comment]
Hi Amir, thanks alot for the codes.
However, is there any Readme? There's too many versions and folders to navigate.

Can you just let me know path of the code on CMU-MOSI and I'll work on the rest.
Thanks

**Comment by A2Zadeh** [comment]
Unfortunately there isn't one yet, I will write that soon. Here is the script to run for CMU-MOSI: /media/bighdd2/paul/attlstm/experiments/mosi/test_mosi_lg.py

If there are any paths to specific files, please let me know and I can share them as soon as possible. Thanks for your patience with our old theano code!

**Comment by sverma88** [comment]
@A2Zadeh, Hi Amir I'm a little lost, can you please confirm as below?

1. So each utterance (a video segment) has a sequence and you've chosen sequence length = 20? Simply it means for CMU-MOSI Train it is 1284*Seq*modality dimension.
2. Y train is (1284*1) as each utterance has a real value sentiment.
3. You've designed MARN and MFN to use the sequence information for any given utterance to better estimate the sentiment, correct? Mathematically for a single utterance, MARN(1*seq*v_dim,1*seq*t_dim,1*seq*a_dim) ----> R i.e. a scalar output.
4. Lastly, I can use data from http://immortal.multicomp.cs.cmu.edu/raw_datasets/processed_data/ to reproduce the model, correct?

**Comment by sverma88** [comment]
@A2Zadeh thanks.

**Comment by sverma88** [comment]
Hi Amir, have you done feature selection on CMU-MOSEI dataset as well?

**Comment by A2Zadeh** [comment]
Hi @sverma88. We don't have the feature selection for MOSEI because the dataset is big, and chances of learning something generalizable from all the features in the feature-set increases. We only do for MOSI due to the dataset being smaller in size.

**Comment by sverma88** [comment]
Hi @A2Zadeh,

I've followed the paper and your code on SDK.
I've implemented my own version of MARN which closely follows MFN structure.
It's available here https://github.com/sverma88/MARN, the main file is MARN.py

Can you please verify the implementation, I've taken LSTMH and MAB from the sdk.

**Comment by A2Zadeh** [comment]
Hi @sverma88. The implementation seems correct to me. The only thing about LSTHM is initializations of biases. There are some tricks to initialize the bias correctly (like forget gate to 1). As far as I can remember I did them correctly on the SDK. But worth double checking with theano code. It's been a long time.

**Comment by sverma88** [comment]
Hi @A2Zadeh, will modify if found discrepancies with Theano code.
Many thanks for the review.
